vizro-core/tests/unit/vizro/models/_components/test_tabs.py

```diff
@@ -0,0 +1,74 @@
+"""Unit tests for vizro.models.Container."""
+import dash_mantine_components as dmc
+import pytest
+from asserts import assert_component_equal
+from dash import html
+
+try:
+    from pydantic.v1 import ValidationError
+except ImportError:  # pragma: no cov
+    from pydantic import ValidationError
+
+import vizro.models as vm
+
+
+@pytest.fixture
+def containers():
+    return [
+        vm.Container(id="container-1", title="Title-1", components=[vm.Button()]),
+        vm.Container(id="container-2", title="Title-2", components=[vm.Button()]),
+    ]
+
+
+class TestTabsInstantiation:
+    """Tests model instantiation and the validators run at that time."""
+
+    def test_create_tabs_mandatory_only(self, containers):
+        tabs = vm.Tabs(id="tab-id", tabs=containers)
+        assert isinstance(tabs.tabs[0], vm.Container) and isinstance(tabs.tabs[1], vm.Container)
```

Comment:
FYI in an ideal world this would be a bit stricter:
```suggestion
        assert all(isinstance(tab, vm.Container) for tab in tabs.tabs) and len(tabs.tabs) == 2
```

Since the current test would pass with `tabs.tabs = [vm.Container(), vm.Container(), something, 123, vm.Container()]` when it shouldn't.

This is the case in lots of our tests and I don't care much, so it's just FYI for your personal enlightenment really - don't bother changing it unless you care a lot.

Reply:
Ah, that's a really good point and one I hadn't considered yet! 👍